lib/ts/recipe/session/sessionClass.ts

```diff
@@ -93,6 +103,10 @@ export default class Session implements SessionContainerInterface {
     // Any update to this function should also be reflected in the respective JWT version
     async mergeIntoAccessTokenPayload(accessTokenPayloadUpdate: any, userContext?: any): Promise<void> {
         const updatedPayload = { ...this.getAccessTokenPayload(userContext), ...accessTokenPayloadUpdate };
+        for (const key of protectedProps) {
+            delete updatedPayload[key];
+        }
```

Comment:
I didn't change this, but I see no big advantage to moving it elsewhere, which would also change how the `regenerateAccessToken` function works, changing the interface.

Reply:
Removing the protected props is a new thing? I suggest that you do not do this check at all in the backend SDK since the core will anyway throw an error.

Reply:
Right - I was looking at the wrong `delete`. Anyway, we do this here because `getAccessTokenPayload` will return the protected props, meaning the check in the core endpoint will always fail. We could:
1. do this here (and we should do it at this level, I think, because I don't think we should change the `regenerateAccessToken` behaviour)
2. provide a `mergeIntoAccessTokenPayload` API on the CDI.

I think 2. is the better solution, but we've been putting it off for a while, and it's not critical.

Reply:
- We can remove the protected props in the regerateAccessToken recipe implementation
- We shuold get rid of the regerateAccessToken function from session index

Reply:
- On second thought, if we are removing the protected prop right before calling the core - and we are OK with that, it's as good as just putting that check in the core itself? I prefer if it's in the core as lesser work for backend sdks.
- If we are silently ignoring protected props for mergeIntoAccessTokenPayload, should we also silently be ignoring during createNewSession / refreshSession? It's strange that in some place it throws an error, and in some place it's ignored.

TO DISCUSS

Reply:
We finally decided to move it into mergeIntoAccessTokenPayload (in sessionClass and in index.ts of this function) before merging the user input into the existing payload